Support non-tag/head refs

MggMuggins opened the pull request:
You can totally just create random references under `.git/refs`, this allows them to be rendered (in orange).

This also only draws `HEAD` when it points to a valid reference (rather than the previous behavior which left a weird floating `HEAD -> refs/heads/main`).

TODOs:
 - [ ] Correctly identify & draw virtual references (any ref can point to another ref, not just `HEAD`)
 - [x] Debug detached head state? (Unknown if this is a bug)
 - [ ] Other special references like `CHERRY_PICK_HEAD`

@gorhawk Just finished my second talk on git using this tool; kudos on this again, I've had fun working with it.

MggMuggins committed:
Support non-tag/head refs

This also only draws HEAD when it points to a valid reference

gorhawk commented on src/index.js:
```diff
@@ -97,9 +97,15 @@ async function main() {
+    // Only draw HEAD if it points to a ref that exists
+    const validHead = some(refData, (ref) => HEAD === ref.name);
+
+    const tagData = filter(refData, (ref) => ref.name.includes("/tags/"));
```
Much better, I didn't even think to use the show-ref command

MggMuggins replied:
IIRC I had to dig for a while to find something that would work here

gorhawk commented on src/index.js:
```diff
@@ -97,9 +97,15 @@ async function main() {
+    const refData = map(compact(splitLines(await showRefs().catch(() => ""))), mapLineToRef);
+
+    // Only draw HEAD if it points to a ref that exists
+    const validHead = some(refData, (ref) => HEAD === ref.name);
```
Here we would lose the feature that displays the detached HEAD:

![image](https://github.com/user-attachments/assets/524ed15d-ea3e-457a-b2c9-b90770657aaf)

MggMuggins replied:
Nice touch, using my example :D

Works now:
![example-repo](https://github.com/user-attachments/assets/45d48560-df57-4347-9338-cd8f29129256)

gorhawk replied:
yeah :D I think this was the only problem so it's good to go

MggMuggins committed:
Fix detached HEAD

git 2.43.0 uses the commit hash in .git/HEAD instead of `ref: ` syntax
when checking out tags, so this only needs to consider commits.

MggMuggins commented:
@gorhawk I'm gearing up for round 3 :grin:

Any chance you'd be willing to merge this?

gorhawk commented:
> @gorhawk I'm gearing up for round 3 😁
> 
> Any chance you'd be willing to merge this?

Sure, it was ok last time and working, if you have anything left just open another one.
